frontend/src/services/http.js

```diff
@@ -0,0 +1,109 @@
+import { useSnackbar } from "notistack"
+import { isExpired } from "../jwt"
+import { useCallback } from "react"
+
+const withJsonBodyIfUpdateAction = (requestOpts) => {
+	const method =
+		requestOpts && requestOpts.method ? requestOpts.method.toLowerCase() : "get"
+
+	if (["post", "put", "patch"].indexOf(method) != -1) {
+		if (!requestOpts) {
+			requestOpts = {}
+		}
+		return {
+			...requestOpts,
+			headers: {
+				...(requestOpts.headers || {}),
+				"Content-Type": "application/json",
+			},
+		}
+	}
+	return requestOpts
+}
+
+export const useHttp = () => {
+	const { enqueueSnackbar } = useSnackbar()
+
+	const withValidateResponse = useCallback(
+		(response) =>
+			response
+				.then((response) => {
+					const contentType = response.headers.get("content-type")
+					if (contentType && !contentType.includes("/json")) {
+						return Promise.reject(
+							new TypeError(
+								"Invalid contentType. Expected json, got " + contentType
+							)
+						)
+					}
+
+					if (!response.ok && !contentType) {
+						return Promise.reject(
+							new Error("Unexpected response: " + response.statusText)
+						)
+					}
+
+					return response
+				})
+				.catch((exception) => {
+					console.error(exception)
+					enqueueSnackbar("Unknown error occurred")
+				}),
+		[enqueueSnackbar]
+	)
+
+	const _fetch = useCallback(
+		(url, requestOpts) =>
+			withValidateResponse(fetch(url, withJsonBodyIfUpdateAction(requestOpts))),
```

Comment:
Some warning from github ficzur:
```
React Hook useCallback has a missing dependency: 'withValidateResponse'. Either include it or remove the dependency array. Outer scope values like 'withJsonBodyIfUpdateAction' aren't valid dependencies because mutating them doesn't re-render the component
```